EIPS/eip-3534.md

```diff
@@ -0,0 +1,310 @@
+Standing on the shoulders of EIP-2930 is only intended to support and further the adoption of next-generation transactions.
+
+### Signature target
+
+The signature signs over the transaction type as well as the transaction data.
+This is done to ensure that the transaction cannot be “re-interpreted” as a transaction of a different type.
+
+## Backwards Compatibility
+
+This EIP requires a hard fork.
+
+## Test Cases
+
+| Segment ID | Block Number | Canonical Block Hash |
+| --- | --- | --- |
+| `e78b1ec3` | `0` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `01e78b1ec3` | `1` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `e4e1c0e78b1ec3` | `15_000_000` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `e8d4a50fffe78b1ec3` | `999_999_999_999` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+| `7fffffffffffffffe78b1ec3` | `9223372036854775807` | `0xe78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58` |
+
+Further test cases, TODO.
+
+## Security Considerations
+
+#### Why 4 bytes of a block hash is "safe enough" for the `ancestorId`
+
+Four bytes, instead of the whole hash (32 bytes), was chosen only to reduce the amount of information required to cross the wire to implement this value.
+Using the whole hash would result in a "perfectly safe" implementation.
+
+The goal of the `ancestorId` is to disambiguate one chain segment from another, and in doing so, enable a transaction to define with adequate precision which chain it needs to be on.
+When a transaction's `ancestorId` references a block, we want to be pretty sure that that reference won't get confused with a different block than the one the author of the transaction had in mind.
+
+Block hashes are effectively unique (at least ethereum/go-ethereum assumes so in their database schema), and are [cited as approximations for random numbers in the yellow paper](https://github.com/ethereum/yellowpaper/blob/41c1837f7b1ddcd65f135763c8e965146cd0ab70/Paper.tex#L1261).
+We assume this trait is uniformly applicable to all possible subsets of the hash value; that is: whether the whole hash, in the first 4 bytes, the first 14, the middle 8, or the last 30; any piece is as random as any other piece. 
+
+- `e78b1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58`. Whole hash, random.
+- `e78b1ec3`. First 4 bytes, random.
+- `e78b1ec31bcb535548ce4b6ef384`. First 14 bytes, random.
+- `384deccad1e7dc59`. Middle 8 bytes, random.
+- `1ec31bcb535548ce4b6ef384deccad1e7dc599817b65ab5124eeaaee3e58`. Last 30 bytes, random.
+
+So our preference of using the _first_ 4 bytes is arbitrary and functionally equivalent to any other subset of equal length.
+
+For the sake of legibility and accessibility, the following arguments will reference the hex representation of 4 bytes, which is 8 characters in length, eg. `e78b1ec3`. 
+
+The chance of a colliding `ancestorId` is:
```

Comment:
I'm having doubts about the correctness of this probability reasoning, thinking I may need some factorials in there. Bear with me.

Reply:
See https://github.com/ethereum/EIPs/pull/3534/commits/265fd8d2f0d6445eb05b1b8a35d07aeb5a0311c5